graphrag/index/workflows/v1/subflows/create_final_communities.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""All the steps to transform final communities."""
+
+from functools import partial
+from typing import Any, cast
+
+import pandas as pd
+from datashaper import (
+    Criterion,
+    FilterArgs,
+    FilterCompareType,
+    StringComparisonOperator,
+    Table,
+    VerbCallbacks,
+    VerbInput,
+    filter_df,
+    verb,
+)
+from datashaper.table_store.types import VerbResult, create_verb_result
+from pandas._typing import Suffixes
+from pandas.api.types import is_bool
+
+from graphrag.index.verbs.graph.unpack import unpack_graph_df
+from graphrag.index.verbs.overrides.aggregate import aggregate_df
+
+
+@verb(name="create_final_communities", treats_input_tables_as_immutable=True)
+def create_final_communities(
+    input: VerbInput,
+    callbacks: VerbCallbacks,
+    **_kwargs: dict,
+) -> VerbResult:
+    """All the steps to transform final communities."""
+    table = cast(pd.DataFrame, input.get_input())
+
+    graph_nodes = unpack_graph_df(table, callbacks, "clustered_graph", "nodes")
+    graph_edges = unpack_graph_df(table, callbacks, "clustered_graph", "edges")
+
+    source_clusters = graph_nodes.merge(
+        graph_edges,
+        left_on="label",
+        right_on="source",
+        how="inner",
+        suffixes=cast(Suffixes, ["_1", "_2"]),
```

Comment:
No need to cast :)

Reply:
Pyright actually complained this time!

Reply:
So odd. This allows for a list of str, even a tuple of str. Ugh... Pyright...